### PR TITLE
July 2020 Progress Report

### PR DESCRIPTION
**Pull request by goldenx86**
Here we go.

### CODE REVIEW

File: site/content/entry/yuzu-progress-report-jul-2020/index.md [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+{{< imgs
+    "./profile.png| Bye little fella!"
+  >}}
+
+## The Sharkman, episode one
+
+[ogniK](https://github.com/ogniK5377) has been working for a long time on a [major audio rewrite](https://github.com/yuzu-emu/yuzu/pull/4310) for yuzu. To avoid a very long explanation of the complexity of Nintendo’s audio renderer systems, we recommend our more tech savvy readers to read the Pull Request’s description where you will find more detailed information.
+ 
+This is “Part 1”, several fixes and more complex (or annoying to code) features will be finished at a later date. Right now the results speak for themselves. Tons of games with broken audio of any kind - like missing sound effects, fading music, mixing errors, crackling or even no sound at all - are either running perfectly or are starting to sound a lot more like the native hardware. Some bugs are to be expected for now, for example, `Shovel Knight` and `Fast RMX` experience choppy audio, and the voices in `Animal Crossing: New Horizons` will play from only one ear.
```
Comment (VolcaEM) [inline site/content/entry/yuzu-progress-report-jul-2020/index.md:170]: I think the "ear" part should be explained better (it's understandable though)
```suggestion
This is “Part 1”, several fixes and more complex (or annoying to code) features will be finished at a later date. Right now, the results speak for themselves. Tons of games with broken audio of any kind - like missing sound effects, fading music, mixing errors, crackling or even no sound at all - are either running perfectly, or are starting to sound a lot more like the native hardware. Some bugs are to be expected for now, for example, `Shovel Knight` and `Fast RMX` experience choppy audio, and the voices in `Animal Crossing: New Horizons` will play from only one ear.
```

File: site/content/entry/yuzu-progress-report-jul-2020/index.md [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+| CPU Threads | Shader Threads | Example CPUs |
+| :-----------: | :--------------: | ------------ |
+| 1 - 7 | 1 | R3-1200, i3-6100 |
+| 8 - 11 | 2 | R5-3300X, i7-7700K |
+| 12 - 15 | 3 | R5-3600, i7-8700K |
+| 16+ | 4 | R7-3700X, i7-10700K |
+
+## Future projects
+
+[Rodrigo](https://github.com/ReinUsesLisp) started rewriting the `Texture Cache`, once the work is finished, this should affect both OpenGL and Vulkan positively, bringing bug fixes, better performance, and a better experience in Vulkan in particular.
```
Comment (VolcaEM) [inline site/content/entry/yuzu-progress-report-jul-2020/index.md:193]:
```suggestion
[Rodrigo](https://github.com/ReinUsesLisp) started rewriting the `Texture Cache`: once the work is finished, this should affect both OpenGL and Vulkan positively, bringing bug fixes, better performance, and a better experience in Vulkan in particular.
```

File: site/content/entry/yuzu-progress-report-jul-2020/index.md [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+    "./mk8bug.png"
+    "./mk8fix.png"
+  >}}
+
+## Bug fixes and improvements
+
+Accompanying the recent addition of 32 bit emulation support, [bunnei](https://github.com/bunnei) implemented [support for the creation of random and default Miis](https://github.com/yuzu-emu/yuzu/pull/4292). This does not include full support for custom ones for now.
+With Miis in place games like `Mario Kart 8 Deluxe` or `New Super Mario Bros. U Deluxe` can now get past their menus and become playable. 
+
+For this service to work, the user needs to dump the system archives from their Nintendo Switch.
```
[the remark this answers is not in the document]
Comment (goldenx86) [reply site/content/entry/yuzu-progress-report-jul-2020/index.md]: Expanded.

File: site/content/entry/yuzu-progress-report-jul-2020/index.md [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+| CPU Threads | Shader Threads | Example CPUs |
+| :-----------: | :--------------: | ------------ |
+| 1 - 7 | 1 | R3-1200, i3-6100 |
+| 8 - 11 | 2 | R5-3300X, i7-7700K |
+| 12 - 15 | 3 | R5-3600, i7-8700K |
+| 16+ | 4 | R7-3700X, i7-10700K |
+
+## Future projects
+
+[Rodrigo](https://github.com/ReinUsesLisp) started rewriting the `Texture Cache`, once the work is finished, this should affect both OpenGL and Vulkan positively, bringing bug fixes, better performance, and a better experience in Vulkan in particular.
```
Comment (Schplee) [inline site/content/entry/yuzu-progress-report-jul-2020/index.md:195]: Change paragraph to:

```Rodrigo started rewriting the Texture Cache and once the work is finished, this should positively affect both OpenGL and Vulkan by introducing bug fixes, better performance, and a better experience in Vulkan in particular.```

File: site/content/entry/yuzu-progress-report-jul-2020/index.md [hunk 5]
```diff
@@ -0,0 +1,209 @@
++++
+date = "2020-08-04T12:00:00-03:00"
+title = "Progress Report July 2020"
+author = "Morph & GoldenX86"
+forum = 0
++++ 
+
+Hey yuz-ers, what is up and welcome back to the… progress report. In this monthly report we offer you major rewrites to critical components, another sliced cut to slow shader compilation, tons of game fixes, and the promise of a very bright future ahead. Let’s get started!
```
[the remark this answers is not in the document]
Comment (goldenx86) [reply site/content/entry/yuzu-progress-report-jul-2020/index.md:8]: Jon wanted some pizza jokes in there.

File: site/content/entry/yuzu-progress-report-jul-2020/index.md [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+
+{{< single-title-imgs
+    "Beware of the dark weeds, legend says rupoors lay there (The Legend of Zelda: Link's Awakening)"
+    "./zlabug.png"
+    "./zlafix.png"
+  >}}
+
+We talked about this in the previous June progress report. [Rodrigo](https://github.com/ReinUsesLisp) finished [implementing the support for `VK_EXT_extended_dynamic_state`](https://github.com/yuzu-emu/yuzu/pull/4150), fixing the “triangle glitch” that affects `Super Mario Odyssey` on all Vulkan drivers that support this new extension.
+
+Thanks to help from [gdkchan](https://github.com/gdkchan) and [Blinkhawk](https://github.com/FernandoS27), [Rodrigo](https://github.com/ReinUsesLisp) added support for [copying 2D arrays of pixels into 3D voxels](https://github.com/yuzu-emu/yuzu/pull/4242), fixing rendering bugs in games like `LEGO Marvel Super Heroes 2` and possibly some others.
```
[the remark this answers is not in the document]
Comment (goldenx86) [reply site/content/entry/yuzu-progress-report-jul-2020/index.md:79]: Those "others" were the fall of RPCS3, they shall never be named.

File: site/content/entry/yuzu-progress-report-jul-2020/index.md [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+
+[toastUnlimited](https://github.com/lat9nq), along with [Rodrigo](https://github.com/ReinUsesLisp), started preliminary work to fix crashes only happening in recent Nvidia Linux Vulkan drivers. Through trial and error, it was found that [increasing the Vulkan device allocable size](https://github.com/yuzu-emu/yuzu/pull/4283) mititagates this issue on most GPUs, but some still remain affected for now.
```
[the remark this answers is not in the document]
Comment (goldenx86) [reply site/content/entry/yuzu-progress-report-jul-2020/index.md:81]: lol

File: site/content/entry/yuzu-progress-report-jul-2020/index.md [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+
+## Project Kobra
+
+Six months ago, [Morph](https://github.com/Morph1984) proposed an [implementation of a batch installer to yuzu](https://github.com/yuzu-emu/yuzu/pull/3385), which allowed users to install many updates and DLCs at the same time. This was a boon for many users, but it remained in early access for a long time due to several issues and anticipation of a new VFS implementation dubbed `Project Viper`. With renewed hope and motivation from the shark [ogniK](https://github.com/ogniK5377), [Morph](https://github.com/Morph1984) went on to fix all of its remaining issues, allowing it to finally be merged into the master/mainline branch of yuzu.
```
[the remark this answers is not in the document]
Comment (goldenx86) [reply site/content/entry/yuzu-progress-report-jul-2020/index.md:91]: @Morph, take responsibility for this section.

File: site/content/entry/yuzu-progress-report-jul-2020/index.md [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+  >}}
+
+## The old guard
+
+When yuzu started using [Conan](https://conan.io/), most of yuzu's dependencies were migrated to it. Unfortunately, one of them, Opus, used for audio decoding, had bad settings in the Conan version, forcing the use of the `AVX/AVX2 instruction set` even on old CPUs that don’t support it, like old AMD Phenom CPUs or any Intel Pentium, Celeron or Atom CPU. This caused some games to crash when processing audio on the affected CPUs. [ogniK](https://github.com/ogniK5377) fixed this by [replacing our included Opus version](https://github.com/yuzu-emu/yuzu/pull/4218).
+
+[Merry](https://github.com/MerryMage) finished the preliminary work needed to [fix a performance regression affecting CPUs that lack the `FMA instruction set`](https://github.com/yuzu-emu/yuzu/pull/4294). This includes AMD Phenom series or older, Intel generation 3 Core i series or older, and all Pentium, Celeron and Atom CPUs. For now this shows up as a debug CPU configuration in yuzu’s settings. We really recommend our users to not play with these settings as altering them will only produce a worse experience.
```
[the remark this answers is not in the document]
Comment (goldenx86) [reply site/content/entry/yuzu-progress-report-jul-2020/index.md:146]: What a stupid name.

File: site/content/entry/yuzu-progress-report-jul-2020/index.md [hunk 10]
```diff
@@ -0,0 +1,209 @@
...
+
+## UI changes
+
+A very requested feature from our users. [toastUnlimited](https://github.com/lat9nq) raised to the challenge and decided to [add per-game configurations](https://github.com/yuzu-emu/yuzu/pull/4098) to yuzu! For now it lacks support for input settings, but this will be added in the future. To access this new feature just right click a game in yuzu's game list and select properties.
```
[the remark this answers is not in the document]
Comment (goldenx86) [reply site/content/entry/yuzu-progress-report-jul-2020/index.md:150]: Are red.

File: site/content/entry/yuzu-progress-report-jul-2020/index.md [hunk 11]
```diff
@@ -0,0 +1,209 @@
...
+
+To top it off, [epicboy](https://github.com/ameerj) added [support for Vulkan](https://github.com/yuzu-emu/yuzu/pull/4443)!
+
+{{< imgs
+    "./async.mp4| Thanks to BSoD Gaming for the video comparison (Super Smash Bros. Ultimate)"
+  >}}
+  
+Due to limitations in the AMD Windows OpenGL driver, we recommend that you avoid using asynchronous shaders in OpenGL if you have a Radeon GPU, as it will only produce frame skipping and lead to a very poor gameplay experience. The Linux OpenGL drivers are unnafected by this, as well as any of the Vulkan drivers.
+
+The number of threads used for asynchronous shader compilation are as follows, along with example CPUs to illustrate the number of CPU threads:
```
[the remark this answers is not in the document]
Comment (goldenx86) [reply site/content/entry/yuzu-progress-report-jul-2020/index.md:184]: Creepypasta

**Pull request merged**